### DISCUSSION TITLE
Limit search scope to a library under the root site

### DISCUSSION THREAD

**Discussion by clmmlc** [microsoft-search/pnp-modern-search#2734] [Q&A]
Hi

I am sure this should be simple but I cannot get the search results to be limited to1 library under our root site.

I have tried different variations of (Path:https://orgname.sharepoint.com/libraryname) but one of them seem to work.

None of the things I have tried retrieve any results. If I remove the query the search works perfectly.

Is there some other configuration I am missing or is my query all wrong?

Any help would be very much appreciated

Thank you

![image](https://user-images.githubusercontent.com/107613292/216234041-06e23c93-bfda-4add-8396-8e5fa73042a4.png)
![image](https://user-images.githubusercontent.com/107613292/216234059-a83e48fe-82f1-4b85-ae08-a49abc0f9e0e.png)
![image](https://user-images.githubusercontent.com/107613292/216234080-783db2ed-5ec3-4410-b8e4-7ad02e479d99.png)

**Comment by fisketryne5** [comment]
Try adding a * after .../SharePointArchive*

**Reply by kasperbolarsen** [reply]
And if you just looking for the library add contentclass=STS_List_DocumentLibrary
and if you just need the documents then IsDocument=1 is what you need :-)

**Reply by clmmlc** [reply]
Thanks so much, I have tried quite a few more iterations of the query and I am still not getting any results, do I need to add brackets or boolean operators to add the content class to get it to work?

{searchTerms}(Path:"https://australianwildlife.sharepoint.com/SharePointArchive*"ContentClass=STS_List_DocumentLibrary)

{searchTerms}Path:"https://orgname.sharepoint.com/SharePointArchive*"ContentClass=STS_List_DocumentLibrary

{searchTerms}ContentClass=STS_List_DocumentLibraryPath:"https://orgname.sharepoint.com/SharePointArchive*"

![image](https://user-images.githubusercontent.com/107613292/216517811-5f99e8a2-251d-4fc4-83e9-d9ce5820c125.png)

**Reply by kasperbolarsen** [reply]
Wait a sec, you have to have a space between the arguments,  so it should be {searchTerms} space Path:

**Reply by clmmlc** [reply]
I've tried it with and without spaces Im afraid, like :

{searchTerms} _space_ Path:"https://orgname.sharepoint.com/SharePointArchive*"ContentClass=STS_List_DocumentLibrary

{searchTerms}_space_Path:"https://orgname.sharepoint.com/SharePointArchive*"_space_ContentClass=STS_List_DocumentLibrary

**Reply by fisketryne5** [reply]
If you remove the librarypath, like "https://orgname.sharepoint.com*", do you get any results at all? 

Have you checked that the library appear in search results?
![image](https://user-images.githubusercontent.com/22908383/216545943-76bb0767-3492-46ce-8745-bb61b494b6f0.png)

Are you sure that the url to the library is correct?

**Reply by wobba** [reply]
This should work if you copy it and edit to your tenant and library URL: `{searchTerms} path:https://orgname.sharepoint.com/sites/mysite/doclib ContentClass:STS_ListItem_DocumentLibrary` The contentclass value you had returns the library instance, not items in the library.

**Reply by clmmlc** [reply]
Thanks @wobba - I have tested it and this works perfectly for my subsites! 

But I think maybe because the library I want it to search is under the root site the URL is shorter than expected? I cannot get it working on a library in the root site

The path to the library becomes shortened to _path:https://orgname.sharepoint.com/doclib ContentClass:STS_ListItem_DocumentLibrary_

**Reply by kasperbolarsen** [reply]
Just tested this using 
path:"https://[tenant].sharepoint.com/sites/SubSiteB/oldSubsite/Shared Documents" ContentClass:STS_ListItem_DocumentLibrary

path:"https://[tenant].sharepoint.com/sites/SubSiteB/Shared Documents" ContentClass:STS_ListItem_DocumentLibrary

path:"https://[tenant].sharepoint.com/Shared Documents" ContentClass:STS_ListItem_DocumentLibrary

All 3 cases work just fine. Could it be that your doclib has a "funny" url ?

**Comment by kasperbolarsen** [comment]
We often ask people to run the query in the SharePoint Search Query Tool https://github.com/pnp/PnP-Tools/releases in order to know if the issue is a generel search API issue or a PnP Modern search issue. Could you try to run the query in the tool ?

**Reply by wobba** [reply]
That or repro with classic search web parts/experience, .e.g via the URL `https://contoso.sharepoint.com/sites/somesite/_layouts/15/osssearchresults.aspx`
